lib/manager/pre-commit/extract.ts

```diff
@@ -0,0 +1,141 @@
+import yaml from 'js-yaml';
+import { forEach } from 'traverse';
+import { id as githubTagsId } from '../../datasource/github-tags';
+import { id as gitlabTagsId } from '../../datasource/gitlab-tags';
+import { logger } from '../../logger';
+import { SkipReason } from '../../types';
+import { regEx } from '../../util/regex';
+import { PackageDependency, PackageFile } from '../common';
+
+import {
+  PrecommitGitDependency,
+  hasKey,
+  matchesPrecommitGitHeuristic,
+} from './util';
+
+function determineDatasource(
+  repository: string,
+  hostName: string
+): { [key: string]: string } {
+  let datasource;
+  let skipReason;
+  if (hostName.endsWith('github.com')) {
```

Comment:
@viceice have we worked out a satisfactory way to do this? The above code is not correct, but I think we'll still get a warning even if we say `if (hostName === 'github.com' || hostName.endsWidth('.github.com'))`. Essentially we want to say `if (domainName === 'github.com')`

Reply:
We should pass `domain` instead of `hostName`, then we can do an equal compare.  Caller function already has extracted `domain`

Reply:
Code has now been changed to actually extract the full domain name, comparison is now done using `===`